include/nbl/builtin/hlsl/radix_fft.hlsl

```diff
@@ -0,0 +1,71 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_HLSL_RADIX_FFT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_RADIX_FFT_INCLUDED_
+
+#include <nbl/builtin/hlsl/complex.hlsl>
+
+
+namespace nbl
+{
+namespace hlsl
+{
+
+
+// Fast Fourier Transform
+template<typename T, uint32_t N>
+void fft_radix2(complex::complex_t<T> data[N], bool is_inverse)
+{
+    const uint32_t log2N = uint32_t(log2(N));
+
+    // Bit reversal permutation
+    for (uint32_t i = 0; i < N; ++i)
+    {
+        uint32_t j = 0;
+        for (uint32_t bit = 0; bit < log2N; ++bit)
+            j |= ((i >> bit) & 1) << (log2N - 1 - bit);
+
+        if (j > i)
+        {
+            complex::complex_t<T> temp = data[i];
+            data[i] = data[j];
+            data[j] = temp;
+        }
+    }
+
+    // Cooley-Tukey
+    for (uint32_t stride = 2; stride <= N; stride *= 2)
+    {
+        complex::complex_t<T> twiddle_factor = { cos((is_inverse ? 2.0 : -2.0) * 3.14 / stride), sin((is_inverse ? 2.0 : -2.0) * 3.14 / stride) };
```

Comment:
use the proper full FP64 value of PI

Reply:
btwa nice piece of trivia @Fletterio , only GLSL defines fp64 transcendentals, HLSL doesn't (the sin,cos etc. functions always return float32, and if you try to use the GL.450.std extended instruction set to compute `cos(double)` this will end up emitting Float64 capability which is a definite DO NOT WANT)

So these definitely need to be tabulated as very precise constants and C-casted into the correct `nbl::hlsl::scalar_type_t<T>` so we don't accidentally emit Float64 capability

Reply:
I'm not really following - we want to use the full PI value in f64, but not emit f64 capability?

Reply:
yes you want to cast the constant into a type you'd be using, see the Gauss Legendre / Quadrature header stuff that Erfan is using for example 62 CAD

Reply:
nvm about the tabulation... its not a perf win compared to just calling sin and cos

for now just assert that `T` for the FFT complex numbers is NOT `float64_t` because that opens a whole can of worms of std.glsl spir-v intrinsic header